sdk/tests/sdk/whirlpools/whirlpool-client-impl.test.ts

```diff
@@ -0,0 +1,113 @@
+import * as anchor from "@project-serum/anchor";
+import * as assert from "assert";
+import Decimal from "decimal.js";
+import {
+  buildWhirlpoolClient,
+  InitPoolParams,
+  PDAUtil,
+  PriceMath,
+  TickUtil,
+  WhirlpoolContext,
+} from "../../../src";
+import { ONE_SOL, systemTransferTx, TickSpacing } from "../../utils";
+import { buildTestPoolParams } from "../../utils/init-utils";
+
+describe("whirlpool-client-impl", () => {
+  const provider = anchor.AnchorProvider.local();
+  anchor.setProvider(anchor.AnchorProvider.env());
+  const program = anchor.workspace.Whirlpool;
+  const ctx = WhirlpoolContext.fromWorkspace(provider, program);
+  const client = buildWhirlpoolClient(ctx);
+
+  let funderKeypair: anchor.web3.Keypair;
+  let poolInitInfo: InitPoolParams;
+  beforeEach(async () => {
+    funderKeypair = anchor.web3.Keypair.generate();
+    await systemTransferTx(provider, funderKeypair.publicKey, ONE_SOL).buildAndExecute();
+    poolInitInfo = (
+      await buildTestPoolParams(
+        ctx,
+        TickSpacing.Standard,
+        3000,
+        PriceMath.priceToSqrtPriceX64(new Decimal(100), 6, 6),
+        funderKeypair.publicKey,
+        false
+      )
+    ).poolInitInfo;
+  });
+
+  it("successfully creates a new whirpool account and initial tick array account", async () => {
```

Comment:
can we also add test cases for the failure cases when token order requires inversion?